### PR TITLE
feat: add move option to a modal window

### PR DESCRIPTION
**Pull request by GlugovGrGlib**
## Description
Adding Move modal functionality

### CODE REVIEW

**Comment by PKulkoRaccoonGang** [comment]
Related FE [PR](https://github.com/raccoongang/frontend-app-course-authoring/pull/236)

File: cms/urls.py [hunk 1]
```diff
@@ -146,8 +146,8 @@
             name='xblock_outline_handler'),
     re_path(fr'^xblock/container/{settings.USAGE_KEY_PATTERN}$', contentstore_views.xblock_container_handler,
             name='xblock_container_handler'),
-    re_path(fr'^xblock/{settings.USAGE_KEY_PATTERN}/editor$', edit_view_xblock,
-            name='xblock_editor_handler'),
+    re_path(fr'^xblock/{settings.USAGE_KEY_PATTERN}/actions/(?P<action_name>[^/]+)$$', xblock_actions_view,
```
Comment (GlugovGrGlib) [inline cms/urls.py:149]: Need to remove $

File: cms/templates/container_editor.html [hunk 2]
```diff
@@ -250,64 +188,24 @@
...
-                        event.preventDefault();
-                        $('.dropdown-options').hide();
-                        XBlockUtils.pasteXBlock($(this)).done(function(data) {
-                            ViewUtils.redirect('/container/' + data.locator + '?action=new');
-                        });
-                    });
-                };
-
-                xblockView.render();
+            require(['js/models/xblock_info', 'js/views/xblock', 'js/views/utils/xblock_utils',
```
Comment (GlugovGrGlib) [inline cms/templates/container_editor.html:192]: Need to remove unused dependencies

File: cms/djangoapps/contentstore/views/block.py [hunk 3]
```diff
@@ -72,8 +73,10 @@
 CREATE_IF_NOT_FOUND = ["course_info"]
 
 # Useful constants for defining predicates
-NEVER = lambda x: False
-ALWAYS = lambda x: True
+def NEVER(x):
```
Comment (GlugovGrGlib) [inline cms/djangoapps/contentstore/views/block.py:76]: Remove formating change

File: cms/static/js/views/modals/edit_xblock.js [hunk 4]
```diff
@@ -208,6 +208,11 @@ function($, _, Backbone, gettext, BaseModal, ViewUtils, XBlockViewUtils, XBlockE
             // Notify child views to stop listening events
             Backbone.trigger('xblock:editorModalHidden');
 
+            window.parent.postMessage({
```
Comment (GlugovGrGlib) [inline cms/static/js/views/modals/edit_xblock.js:211]: Send message for edit

File: cms/static/js/views/modals/move_xblock_modal.js [hunk 5]
```diff
@@ -185,6 +185,10 @@ function($, Backbone, _, gettext, BaseView, XBlockViewUtils, MoveXBlockUtils, Ht
                     targetParentLocator: this.targetParentXBlockInfo.id
                 }
             );
+            window.parent.postMessage({
```
Comment (GlugovGrGlib) [inline cms/static/js/views/modals/move_xblock_modal.js:188]: Send message for move

File: cms/static/js/views/modals/base_modal.js [hunk 6]
```diff
@@ -119,6 +119,10 @@ define(['jquery', 'underscore', 'gettext', 'js/views/baseview'],
                     event.preventDefault();
                     event.stopPropagation(); // Make sure parent modals don't see the click
                 }
+                window.parent.postMessage({
+                    method: 'close_edit_modal',
```
Comment (GlugovGrGlib) [inline cms/static/js/views/modals/base_modal.js:123]: This message is incorrect, it should be generic for modals

File: cms/djangoapps/contentstore/views/block.py [hunk 7]
```diff
@@ -301,20 +302,25 @@ def xblock_view_handler(request, usage_key_string, view_name):
         return HttpResponse(status=406)
 
 
+@xframe_options_exempt
 @require_http_methods("GET")
 @login_required
-def edit_view_xblock(request, usage_key_string):
+def xblock_actions_view(request, usage_key_string, action_name):
     """
     The handler for rendered edit xblock view.
```
Comment (GlugovGrGlib) [inline cms/djangoapps/contentstore/views/block.py:310]: Please update Docstring to describe the rendering for different xblock actions

"""
Return rendered xBlock action view. 
The action name should be provided as an argument.
Valid options for action name are edit and move.
"""

File: cms/static/js/views/modals/edit_xblock.js [hunk 8]
```diff
@@ -208,6 +208,11 @@ function($, _, Backbone, gettext, BaseModal, ViewUtils, XBlockViewUtils, XBlockE
...
+              method: 'close_edit_modal',
+              msg: 'Sends a message when the modal window is closed'
```
Comment (GlugovGrGlib) [inline cms/static/js/views/modals/edit_xblock.js:213]: Please update message to specifically state about edit modal

**Pull request merged**